app/client/src/pages/Editor/DataSourceEditor/NewActionButton.tsx

```diff
@@ -59,14 +75,7 @@ function NewActionButton(props: NewActionButtonProps) {
 
   const createQueryAction = useCallback(
     (pageId: string) => {
-      if (
-        pluginType === PluginType.API &&
-        (!datasource ||
-          !datasource.datasourceStorages[currentEnvironment]
-            .datasourceConfiguration ||
-          !datasource.datasourceStorages[currentEnvironment]
-            .datasourceConfiguration.url)
-      ) {
+      if (isApiPluginInvalidUrl(currentEnvironment, pluginType, datasource)) {
```

Comment:
As we check for presence of URL here, could we name the method accordingly? 
```suggestion
      if (apiPluginHasUrl(currentEnvironment, pluginType, datasource)) {
```